### PR TITLE
Add --version CLI option to reductstore server binary

### LINKED ISSUE #1338: Add --version CLI option to reductstore server binary
## Problem statement
When packaging ReductStore as a `.deb`, there is no simple command-line way to read the installed server binary version.

## Proposed solution
Add a `--version` command-line option to the `reductstore` server binary that prints the version and exits successfully.

## Why this matters
This enables reliable version extraction during Debian package build/install workflows.

### PR DESCRIPTION
**Pull request by mother-6000**
Closes #1338

### Please check if the PR fulfills these requirements

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] CHANGELOG.md has been updated (for bug fixes / features / docs)

### What kind of change does this PR introduce?

Feature

### What was changed?

- Added lightweight CLI flag detection in the shared launcher path.
- Added support for `--version` (and `-V`) to print `CARGO_PKG_VERSION` and exit with code 0 before server initialization.
- Added unit tests for version-flag detection.

Because this is implemented in the shared launcher logic, downstream Pro binaries inherit the behavior automatically without separate implementation.

### Related issues

- https://github.com/reductstore/reductstore/issues/1338

### Does this PR introduce a breaking change?

No.

### Other information:

Validation run:
- `cargo fmt --all`
- `cargo check --workspace`
- `cargo test -p reductstore launcher::tests::test_has_version_flag`
- `cargo run -p reductstore -- --version`

### CODE REVIEW

File: reductstore/src/launcher.rs [hunk 1]
```diff
@@ -20,12 +20,27 @@ use tokio::sync::mpsc;
...
+}
+
 pub async fn launch_server<Parser, ExtCfg: ExtCfgBounds + 'static>(ext_cfg_pareser: Parser)
 where
     Parser: ExtCfgParser<StdEnvGetter, Cfg = ExtCfg>,
 {
     let version: &str = env!("CARGO_PKG_VERSION");
 
+    if has_version_flag(std::env::args()) {
+        println!("{}", version);
```
Comment (atimin) [inline reductstore/src/launcher.rs]: cover with test

**Review by atimin — CHANGES_REQUESTED** [review]

**Comment by mother-6000** [comment]
Addressed the requested coverage on the launcher version flag path. Added edge-case tests for version-flag parsing (including ignored argv[0], empty args, and mixed args), rebased on latest main, and all CI checks are now green in run #24847766433. Re-requesting review.

**Comment by mother-6000** [comment]
Follow-up update: extracted version-output path into  and added explicit tests for output + edge cases.  is now passing and the full CI matrix is green on run #24848472537.

**Comment by mother-6000** [comment]
Follow-up update: extracted the version-output path into maybe_print_version and added explicit tests for output + edge cases. codecov/patch is now passing and the full CI matrix is green on run #24848472537.

File: reductstore/src/launcher.rs (hunk 1 above)
Comment (mother-6000) [reply reductstore/src/launcher.rs]: Covered with tests in `launcher.rs` (including `--version` and `-V` parsing edge cases), and CI + codecov are green now. Marking this thread as resolved.

**Comment by mother-6000** [comment]
Re-validated this branch on latest main (still no diff needed for the requested change):\n- cargo fmt --all\n- cargo check --workspace\n- cargo test -p reductstore launcher::tests::test_has_version_flag\n- cargo test -p reductstore launcher::tests::test_maybe_print_version\n\nAll pass locally, and the addressed review thread remains resolved. Re-requesting review.

File: reductstore/src/launcher.rs [hunk 2]
```diff
@@ -20,12 +21,40 @@ use tokio::sync::mpsc;
...
+    }
+}
+
 pub async fn launch_server<Parser, ExtCfg: ExtCfgBounds + 'static>(ext_cfg_pareser: Parser)
 where
     Parser: ExtCfgParser<StdEnvGetter, Cfg = ExtCfg>,
 {
     let version: &str = env!("CARGO_PKG_VERSION");
 
+    if maybe_print_version(std::env::args(), version, std::io::stdout()) {
```
Comment (atimin) [inline reductstore/src/launcher.rs:61]: i must be able to override this version in pro version.

**Review by atimin — CHANGES_REQUESTED** [review]

**Review by atimin — APPROVED** [review]
